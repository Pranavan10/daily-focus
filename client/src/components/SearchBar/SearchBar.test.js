import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";
import Paper from "@material-ui/core/Paper";
import IconButton from "@material-ui/core/IconButton";
import InputBase from "@material-ui/core/InputBase";

import SearchBar from "./";

let component;

beforeEach(() => {
    component = shallow(<SearchBar />);
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<SearchBar />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("it contains Material UI Paper, InputBase, and IconButton components", () => {
    expect(component.find(Paper)).toHaveLength(1);
    expect(component.find(InputBase)).toHaveLength(1);
    expect(component.find(IconButton)).toHaveLength(1);
});
