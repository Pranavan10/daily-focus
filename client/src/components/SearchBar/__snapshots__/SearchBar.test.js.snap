// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  action="https://www.google.co.nz/search"
  classes={
    Object {
      "root": "makeStyles-searchBar-4",
    }
  }
  component="form"
  elevation={3}
  method="get"
  target="_blank"
>
  <div
    className="container"
  >
    <img
      alt="Google"
      className="googleLogo"
      src="google-icon.svg"
    />
    <WithStyles(ForwardRef(InputBase))
      classes={
        Object {
          "input": "makeStyles-searchLabel-2",
          "root": "makeStyles-root-1",
        }
      }
      inputComponent="input"
      inputProps={
        Object {
          "aria-label": "search google",
          "name": "query",
          "type": "search",
        }
      }
      placeholder="Google Search"
    />
    <WithStyles(ForwardRef(IconButton))
      aria-label="search"
      classes={
        Object {
          "root": "makeStyles-searchIcon-3",
        }
      }
      type="submit"
    >
      <Memo />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</WithStyles(ForwardRef(Paper))>
`;
